tests/test_update.py

```diff
@@ -195,3 +196,25 @@ def test_merge_deprecation() -> None:
     with warns(UserWarning, match=re.escape(msg)):
         OmegaConf.update(cfg, "a", {"c": 20})  # default to set, and issue a warning.
         assert cfg == {"a": {"c": 20}}
+
+
+@mark.parametrize(
+    "cfg,key,value,expected",
+    [
+        param({}, "a", 10, {"a": 10}, id="add_value"),
+        param({}, "a.b", 10, {"a": {"b": 10}}, id="add_value"),
+        param({}, "a", {"b": 10}, {"a": {"b": 10}}, id="add_dict"),
+        param({}, "a.b", {"c": 10}, {"a": {"b": {"c": 10}}}, id="add_dict"),
+        param({}, "a", [1, 2], {"a": [1, 2]}, id="add_list"),
+        param({}, "a.b", [1, 2], {"a": {"b": [1, 2]}}, id="add_list"),
+    ],
+)
+def test_update_force_add(cfg: Any, key: str, value: Any, expected: Any) -> None:
+    cfg = _ensure_container(cfg)
+    OmegaConf.set_struct(cfg, True)
```

Comment:
Can you add a test where the struct node isn't the root, but rather a node in the middle of the update path?

Reply:
Added something similar with a Structured Config.